ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+                                                'write_lan': params['domains']['lan']['write'],
+                                                'write_wan': params['domains']['wan']['write'],
+                                                'delete_lan': params['domains']['lan']['delete'],
+                                                'delete_wan': params['domains']['wan']['delete'],
+                                                'third_party_copy': params['domains']['wan']['write']}
+                                        self.log.info('Will update protocol %s://%s:%s%s on %s' % (params['scheme'], params['hostname'], params['port'], params['prefix'], se))
+                                        client.update_protocols(rse=se, scheme=params['scheme'], data=data, hostname=params['hostname'], port=params['port'])
+                    for prot in existing_protocols:
+                        if prot not in protocols_to_create:
+                            self.log.info('Will delete protocol %s://%s:%s%s on %s' % (prot[0], prot[1], prot[2], prot[3], se))
+                            client.delete_protocols(se, scheme=prot[0], hostname=prot[1], port=prot[2])
+            else:
+                self.log.error('Cannot get SEs : %s' % result['Value'])
+
+            # If new RSE added, add distances
+            rses = [rse['rse'] for rse in client.list_rses()]
+            if newRSE:
+                self.log.info("Adding distances")
+                for src_rse in rses:
+                   for dest_rse in rses:
+                       if src_rse != dest_rse:
+                           try:
+                               client.add_distance(src_rse, dest_rse, {'ranking': 1, 'distance': 10})
+                           except Exception as err:
+                               self.log.error('Cannot add distance for %s:%s : %s' % (src_rse, dest_rse, str(err)))
+
+            # Collect the shares from Dirac Configuration and create them in Rucio
+            self.log.info("Synchronizing shares")
+            result = Operations().getOptionsDict('Production/SEshares')
+            if result['OK']:
+                rseDict = result['Value']
+                for rse in rses:
+                    try:
+                        self.log.info('Setting productionSEshare for %s : %s' % (rse, rseDict.get(rse, 0)))
+                        client.add_rse_attribute(rse, 'productionSEshare', rseDict.get(rse, 0))
+                    except Exception as err:
+                        self.log.error('Cannot create productionSEshare for %s' % rse)
+            else:
+                self.log.error('Cannot get SEs : %s' % result['Value'])
+
+            result = Operations().getSections('Shares')
+            if result['OK']:
+                for dataLevel in result['Value']:
+                    result = Operations().getOptionsDict('Shares/%s' % dataLevel)
+                    if result['OK']:
```

Comment:
Another example where `continue` can reduce indentation and make it easier to follow

Reply:
I don't understand the comment

Reply:
Something like this to avoid indenting any more deeply than needed:
```python
if not result['OK']:
  self.log.error('Cannot get SEs : %s' % result['Value'])
  continue
attribute = '%sShare' % dataLevel
for rse in rses:
  value = result['Value'].get(rse, 0)
  self.log.info('Setting %s for %s : %s' % (attribute, rse, value))
  try:
    client.add_rse_attribute(rse, attribute, value)
  except Exception as err:
    self.log.error('Cannot create %s for %s' % (attribute, rse))
```

Reply:
Also I guess the logging should used the fixed/variable split like:
```diff
- self.log.error('Cannot create %s for %s' % (attribute, rse))
+ self.log.error('Cannot create RSE attribute: ', '%s for %s' % (attribute, rse))
```

Reply:
Ah, right thanks for the clarification

Reply:
Actually, we don't want to continue here because other things are done after and we don't want to skip them

Reply:
Nothing else is done in the `for dataLevel in result['Value']:` loop? Or am I missing something?

Reply:
Yes, you're right. Done